modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -995,26 +1004,29 @@ virtual Ptr<BackendNode> initNgraph(const std::vector<Ptr<BackendWrapper> >& inp
         std::vector<int> inpShape(inputs[0].begin() + 2, inputs[0].end());
         std::vector<int> outShape(inputs[0].begin(), inputs[0].begin() + 2);
 
-        if (globalPooling)
-        {
-            outShape.push_back(1);
-            outShape.push_back(1);
+        std::vector<size_t> local_kernel = kernel_size.empty() ?
+                                           std::vector<size_t>(inpShape.begin(), inpShape.end()) : kernel_size;
+
+        for (int i = 0, j = local_kernel.size() - inpShape.size(); i < inpShape.size(); i++, j++) {
+            if (isGlobalPooling[j])
+                local_kernel[j] = inpShape[i];
```

Comment:
Can we move it under `if (globalPooling)` so that will guarantee that kernel is not empty?